### DISCUSSION TITLE
Code coverage report

### DISCUSSION THREAD

**Discussion by connorhu** [FriendsOfSymfony1/symfony1#322] [Ideas]
Do we want codecov integration where it is possible? For now, swiftmailer uses phpunit, but doctrine1 and symfony1 can be adapted in the foreseeable future. For symfony1 I would need php7.4 minimum to start integrating phpunit.

**Comment by alquerci** [comment]
My opinions about code coverage:

It is useful only for mutation testing, in order to characterize the code without proof of its behavior.
Make a golden master.

So for me codecov is useless.

**Reply by connorhu** [reply]
If not for the sake of chasing a 100% metric, but it gives a realistic picture of the code, whether there is still deprecation warning lurking in a file. If a file is not tested at all, it may be worth looking at. We don't really have a clear picture of that right now.

**Reply by thePanz** [reply]
@connorhu Symfony1 is already set to PHP v7.4 as the lowest version now (See #325)

**Reply by connorhu** [reply]
Current problem: We got a PR that changes a line in a file. In order to check that the change doesn't cause a BC break, I need to search out the tests for that section and look through what the tests test, one by one. Using code coverate report, I just check if that is tested and I know right away that I don't need to ask the PR owner for a test, just green status is enough.

**Reply by alquerci** [reply]
@connorhu for that kind of use case you describe. Code coverage does not tell you if new code doesn't cause a BC break.

Mutation testing is more useful to chase BC break.

Code coverage is one step toward mutation testing.
I agree to add code coverage only on the context of mutation testing report.
